tools/src/main/scala/scala/scalanative/optimizer/pass/Canonicalization.scala

```diff
@@ -0,0 +1,52 @@
+package scala.scalanative
+package optimizer
+package pass
+
+import analysis.ClassHierarchy._
+import nir._, Inst.Let
+
+/** Moves constant operands in Op.Bin and Op.Comp to the right. */
+class Canonicalization extends Pass {
+
+  override def preInst = {
+    case Let(n, Op.Bin(bin, ty, lhs, rhs))
+        if (commutativeBin(bin) && constant(lhs)) =>
+      Seq(Let(n, Op.Bin(bin, ty, rhs, lhs)))
+
+    case Let(n, Op.Comp(comp, ty, lhs, rhs))
+        if (commutativeComp(comp) && constant(lhs)) =>
+      Seq(Let(n, Op.Comp(comp, ty, rhs, lhs)))
+  }
+
+  private def commutativeBin(bin: Bin): Boolean = {
+    import Bin._
+    bin match {
+      case Iadd | Fadd | Imul | Fmul | And | Or | Xor => true
+      case Isub | Fsub | Sdiv | Udiv | Fdiv | Srem | Urem | Frem | Shl | Lshr |
+          Ashr =>
+        false
+    }
+  }
+
+  private def commutativeComp(comp: Comp): Boolean = {
+    import Comp._
+    comp match {
+      case Ieq | Ine | Feq | Fne => true
```

Comment:
Does the previous comment about floating point ops mean that we also don't do anything about `Feq` and `Fne` ?